packages/gatsby-mdx/mdx-renderer.js

```diff
@@ -2,17 +2,18 @@ import { withMDXComponents } from "@mdx-js/tag/dist/mdx-provider";
 import { withMDXScope } from "./context";
 
 export default withMDXScope(
-  withMDXComponents(({ scope = {}, components = {}, children, ...props }) => {
```

Comment:
The `MDXRenderer` currently pulls in all the scopes instead of just the one needed for the code. This change makes it only pull in the exact scope for the code.

Instead of taking in just the `body` from the `code` field, it now takes in the whole `code` field so it has the scope id to pull in.